qt/widgets/common/src/SequentialFitDialog.cpp

```diff
@@ -373,7 +372,61 @@ void SequentialFitDialog::functionChanged() { populateParameters(); }
 
 void SequentialFitDialog::finishHandle(
     const Mantid::API::IAlgorithm * /*alg*/) {
+  getFitResults();
   emit needShowPlot(&ui, m_fitBrowser);
+  m_fitBrowser->sequentialFitFinished();
+}
+
+/// Set the parameters to the fit outcome.
+/// if more than one entry in the gui the parameters used are first entry with
+/// same name as fit property browser if one entry will try and use row that
+/// corresponds to ws index in fit browser, if not found defaults to first row.
+void SequentialFitDialog::getFitResults() {
+  auto wsName = m_outputName;
+  if (Mantid::API::AnalysisDataService::Instance().doesExist(wsName)) {
+    Mantid::API::ITableWorkspace_sptr ws =
+        boost::dynamic_pointer_cast<Mantid::API::ITableWorkspace>(
+            Mantid::API::AnalysisDataService::Instance().retrieve(wsName));
+
+    // first column is ws name or axis no
+    // the columns after are pairs of parameter values and errors
+    // the last column is chi-squared
+
+    auto columnNames = ws->getColumnNames();
+
+    int rowNo = 0;
+    if (rowCount() > 1) {
+      // contains ws names
+      auto firstColumn = ws->getColumn(0);
+      for (size_t i = 0; i < ws->rowCount(); ++i) {
+        if (firstColumn->cell<std::string>(i) == m_fitBrowser->workspaceName())
+          rowNo = static_cast<int>(i);
```

Comment:
Is it worth breaking after we find this, or are would there be multiple cells with the workspace name and were looking for the very last?

Reply:
Good spot, I'll add this in. There could be multiple cells with the same name but we want the first.